src/common/Requirements.ts

```diff
@@ -86,54 +88,64 @@ export abstract class BaseSetup implements RequirementList {
         ];
     }
     public async executeSetup(): Promise<SetupTestResult> {
-        const allPromises: Array<Promise<any>> = [];
-        this.requirements.forEach((requirement) =>
-            allPromises.push(WrappedPromise(requirement.checkFunction()))
-        );
-        const logger = this.logger;
-        const reqs = this.requirements;
-        return Promise.all(allPromises).then((results) => {
-            const testResult: SetupTestResult = {
-                hasMetAllRequirements: true,
-                tests: []
-            };
-            let count = 0;
-            results.forEach((result) => {
-                if (result.status === 'fulfilled') {
-                    testResult.tests.push({
-                        hasPassed: true,
-                        message: result.v,
-                        testResult: 'Passed'
-                    });
-                } else if (result.status === 'rejected') {
-                    testResult.hasMetAllRequirements = false;
-                    testResult.tests.push({
-                        hasPassed: false,
-                        message: result.e,
-                        testResult: 'Failed'
-                    });
-                }
-                count++;
-            });
-
-            const tree = cli.tree();
-            tree.insert('Setup');
-            testResult.tests.forEach((test) => {
-                tree.nodes.Setup.insert(
-                    `${
-                        test.hasPassed
-                            ? chalk.bold.green(test.testResult)
-                            : chalk.bold.red(test.testResult)
-                    }: ${
-                        test.hasPassed
-                            ? chalk.green(test.message)
-                            : chalk.red(test.message)
-                    }`
-                );
-            });
-            tree.display();
-            return Promise.resolve(testResult);
+        const testResult: SetupTestResult = {
+            hasMetAllRequirements: true,
+            tests: []
+        };
+
+        let totalDuration: number = 0;
+        let stepDuration: number = 0;
+        const obs = new PerformanceObserver((items) => {
+            stepDuration = items.getEntries()[0].duration / 1000;
+            performance.clearMarks();
```

Comment:
It may feel less disconnected to update `totalDuration` here. Same difference, but it keeps the processes together a little better, IMO.